mlir/lib/Dialect/Rock/Transforms/GridwiseGemmToBlockwise.cpp

```diff
@@ -1167,508 +1167,138 @@ struct GridwiseGemmV2RewritePattern
     uint32_t blockSize = op.getBlockSize();
     uint32_t gridSize = op.getGridSize();
     XdlopsGemmParamsAttr tuningParams = op.getParams();
-    int64_t KPack = tuningParams.getKpack();
-    int64_t KPerBlock = tuningParams.getKPerBlock();
-    int64_t MPerBlock = tuningParams.getMPerBlock();
-    int64_t NPerBlock = tuningParams.getNPerBlock();
-
-    int64_t matrix_a_source_data_per_read = 0;
-    int64_t matrix_b_source_data_per_read = 0;
-    GemmDimension matrix_a_source_vector_read_dim;
-    GemmDimension matrix_b_source_vector_read_dim;
-
-    int64_t aCopyPerThread = (KPerBlock * KPack * MPerBlock) / blockSize;
-    int64_t bCopyPerThread = (KPerBlock * KPack * NPerBlock) / blockSize;
+    int64_t kpack = tuningParams.getKpack();
+    // TODO: kPerBlock, as defined in parameter selection etc,
+    // is in units of kPack, not individual k. This should be changed
+    // at some future point, but it'll be worked around for now.
+    int64_t kpacksPerBlock = tuningParams.getKPerBlock();
+    int64_t mPerBlock = tuningParams.getMPerBlock();
+    int64_t nPerBlock = tuningParams.getNPerBlock();
+    int64_t mBlocks = M / mPerBlock;
+    int64_t nBlocks = N / nPerBlock;
+
+    int64_t kPerBlock = kpacksPerBlock * kpack;
+
+    int64_t aVectorLen = 0;
+    int64_t bVectorLen = 0;
+    GemmDimension aVectorDim;
+    GemmDimension bVectorDim;
+
+    SmallVector<StringRef, 3> bidGridOrder = {"g_block", "m_block", "n_block"};
```

Comment:
Potential bug source: the xdlops code expects "m_block" and "n_block" to be the other way around.

Reply:
That was some amazing bug-finding abilities here. Thanks so much and this is it.

Reply:
It's less "amazing bug-finding abilities" and more:
1. I ran into this weird quirk back when I was refactoring the output loops and
2. Remembered it would be coming up when I did the non-xdlops refactor, so I planned for it and
3. When I saw this line, it reminded me of the possibility of this bug and so I went off to check if this could be the problem